docs/contracts/transient-storage.rst

```diff
@@ -0,0 +1,166 @@
+.. index:: ! transient storage, ! transient, tstore, tload
+
+.. _transient-storage:
+
+*****************
+Transient Storage
+*****************
+
+Transient storage is another data location besides memory, storage, calldata
+(and return-data and code) which was introduced alongside its respective opcodes
+``TSTORE`` and ``TLOAD`` by `EIP-1153 <https://eips.ethereum.org/EIPS/eip-1153>`_.
+This new data location behaves as a key-value store similar to storage with the main
+difference being that data in transient storage is not permanent, but is scoped to
+the current transaction only, after which it will be reset to zero.
+Since the content of transient storage has very limited lifetime and size,
+it does not need to be stored permanently as a part of state
+and the associated gas costs are much lower than in case of storage.
+EVM version ``cancun`` or newer is required for transient storage to be available.
+
+Transient storage variables cannot have an initializer since the value would be cleared at
```

Comment:
```suggestion
Transient storage variables cannot be initialized in place since the value would be cleared at
```

Reply:
I also think it would be useful to add a sentence saying that all transient state variables will be default initialized to whatever is the default for their type, and maybe link [this](https://docs.soliditylang.org/en/latest/control-structures.html#scoping-and-declarations) somewhere as well.

Reply:
I also added a clarification next to `initialized in place` so to make it very obvious what we refer to.